go/libraries/doltcore/sqle/enginetest/dolt_queries.go

```diff
@@ -705,6 +705,65 @@ var DoltRevisionDbScripts = []queries.ScriptTest{
 // DoltScripts are script tests specific to Dolt (not the engine in general), e.g. by involving Dolt functions. Break
 // this slice into others with good names as it grows.
 var DoltScripts = []queries.ScriptTest{
+	{
+		Name: "test has_ancestor",
```

Comment:
Should also test tags, since it's the motivating customer use case

Also add some tests for e.g. `use mydb/tag1`, `use mydb/HEAD^2`